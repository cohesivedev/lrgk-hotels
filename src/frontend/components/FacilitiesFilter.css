.FacilitiesFilter {
    border: 1px solid black;
    padding: 1em;
    margin-bottom: 1em;
    line-height: 1.5em;
}

.FacilitiesFilter label {
    margin: 0.5em;
    padding: 0 1em;
    display: inline-block;
    border: 3px solid lightgray;
}

.FacilitiesFilter input {
    display: none;
}

.FacilitiesFilter label.selected {
    border: 3px solid blue;
    border-radius: 0.2em;
}

.FacilitiesFilter .actions {
    text-align: right;
}

.FacilitiesFilter .actions button {
    padding: 1em;
}