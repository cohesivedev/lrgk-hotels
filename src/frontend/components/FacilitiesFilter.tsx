import * as React from 'react';
import {ChangeEventHandler, MouseEventHandler} from 'react';

import './FacilitiesFilter.css';

interface IFacilitiesFilterProps {
    validFilters: string[];
    activeFilters: string[]
    onChange: ChangeEventHandler;
    onClickClear: Function;
}

const FacilitiesFilter: React.FC<IFacilitiesFilterProps> = ({validFilters, activeFilters, onChange, onClickClear}) => (
    <div className="FacilitiesFilter">
        <div>
            Show me hotels that include
        </div>
        {validFilters.map(f => {
            const checked = activeFilters.includes(f);

            return (
                <label key={`FacilitiesFilter.${f}`} className={checked ? 'selected' : ''}>
                    {f}

                    <input
                        onChange={onChange}
                        type="checkbox"
                        data-filter={f}
                        checked={checked}
                    />
                </label>
            );
        })}
        <div className="actions">
            <button onClick={onClickClear as MouseEventHandler}>Clear all</button>
        </div>
    </div>
);

export default FacilitiesFilter;