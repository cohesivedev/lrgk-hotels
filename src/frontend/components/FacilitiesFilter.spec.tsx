import * as React from 'react';
import FacilitiesFilter from './FacilitiesFilter';
import {renderToString} from 'react-dom/server';

describe('components/FacilitiesFilter', () => {
    it('should match snapshot',
        () => expect(
            renderToString(
                <FacilitiesFilter
                    validFilters={'ABC'.split('')}
                    activeFilters={'AC'.split('')}
                    onChange={new Function() as any}
                />
            )
        ).toMatchSnapshot()
    );
});
