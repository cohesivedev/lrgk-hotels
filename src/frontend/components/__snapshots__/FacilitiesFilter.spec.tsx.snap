// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FacilitiesFilter should match snapshot 1`] = `

<div class="FacilitiesFilter"
     data-reactroot
>
  <div>
    Show me hotels that include
  </div>
  <label class="selected">
    A
    <input type="checkbox"
           data-filter="A"
           checked
    >
  </label>
  <label class>
    B
    <input type="checkbox"
           data-filter="B"
    >
  </label>
  <label class="selected">
    C
    <input type="checkbox"
           data-filter="C"
           checked
    >
  </label>
  <div class="actions">
    <button>
      Clear all
    </button>
  </div>
</div>

`;
